core/block/simple/dataview/diff_test.go

```diff
@@ -0,0 +1,1126 @@
+package dataview
+
+import (
+	"github.com/anyproto/anytype-heart/pb"
+	"github.com/anyproto/anytype-heart/pkg/lib/pb/model"
+	"github.com/anyproto/anytype-heart/util/pbtypes"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestDiff(t *testing.T) {
+	testBlock := func() *Dataview {
+		return NewDataview(&model.Block{
+			Restrictions: &model.BlockRestrictions{},
+			Content:      &model.BlockContentOfDataview{Dataview: &model.BlockContentDataview{}},
+		}).(*Dataview)
+	}
+	t.Run("is collection changed", func(t *testing.T) {
+		// given
+		b1 := testBlock()
+		b2 := testBlock()
+
+		// when
+		b2.content.IsCollection = true
+		diff, err := b1.Diff(b2)
+
+		// then
+		require.NoError(t, err)
+		require.Len(t, diff, 1)
+		change := diff[0].Msg.Value.(*pb.EventMessageValueOfBlockDataviewIsCollectionSet).BlockDataviewIsCollectionSet
```

Comment:
I think we should use more declarative-style assertions, something like this

```
		assert.Equal(t, makeEvents(
			&pb.EventMessageValueOfBlockDataviewIsCollectionSet{
				BlockDataviewIsCollectionSet: &pb.EventBlockDataviewIsCollectionSet{
					Value: true,
				},
			}), diff)
```

where makeEvents is helper for building []simple.EventMessage


Such approach will lead to a couple of positive effects:
1. Tests will become more comprehensible and less error-prone
2. If Diff will produce changes in other fields then test shows it immediately

Reply:
Thank you, fixed